app/letters/utils.py

```diff
@@ -5,24 +5,32 @@
 
 from notifications_utils.s3 import s3upload
 
+from app.models import KEY_TYPE_TEST
 from app.variables import Retention
 
 
 LETTERS_PDF_FILE_LOCATION_STRUCTURE = \
-    '{folder}/NOTIFY.{reference}.{duplex}.{letter_class}.{colour}.{crown}.{date}.pdf'
+    '{folder}NOTIFY.{reference}.{duplex}.{letter_class}.{colour}.{crown}.{date}.pdf'
 
-PRECOMPILED_BUCKET_PREFIX = '{folder}/NOTIFY.{reference}'
+PRECOMPILED_BUCKET_PREFIX = '{folder}NOTIFY.{reference}'
 
 
-def get_letter_pdf_filename(reference, crown):
-    now = datetime.utcnow()
+def get_folder_name(_now, is_test_letter):
+    if is_test_letter:
+        folder_name = ''
+    else:
+        print_datetime = _now
```

Comment:
Should we use this as an opportunity to use created_at instead of now()?

It would be less prone to error and would mean we haven't got to search the bucket but use the created_at date. Get file would be just obj = s3.Object(bucket_name=bucket_name, key=item.key)

Technically a notification could be created at <deadline>:999 and this be ran on <deadline>:001 which would cause the folder name to be different for the get_letter_pdf function

Reply:
Wanted to keep this PR as small as possible, that other issue could be fixed in another PR

Reply:
Had another thought on this, if the letter gets processed but then we create the pdf after the cut off time, there's a potential that it might never gets processed if it's using the `created_at` time to indicate which subfolder it will be placed in. Think we need a bit more discussion around this.